### ISSUE TITLE
thespruceeats missing ingredients

### ISSUE THREAD

**Issue by smilerz** [hhursev/recipe-scrapers#383]
It looks like the thespruceeats.com uses 2 different formats to list ingredients.  Both the current method used by the scraper:

```
ingredients = self.soup.find(
            "section",
            {"class": "comp section--fixed-width section--ingredients section"},
        ).find_all("li", {"class": "structured-ingredients__list-item"})
```

as well as the prior method that is currently commented out.

`# ingredients = self.soup.findAll("li", {"class": "ingredient"})`

**Pre-filing checks**

- [x] I have searched for open issues that report the same problem
- [x] I have checked that the bug affects the latest version of the library

**The URL of the recipe(s) that are not being scraped correctly**

- https://www.thespruceeats.com/creamy-potato-soup-with-ham-3059797

**The version of Python you're using**
3.9

**The operating system of your environment**
Debian

**The results you expect to see**
list of ingredients

**The results (including any Python error messages) that you are seeing**
empty list

Can you write Python and would you like to help fix the scraper yourself?  We'd be glad for your assistance!  We can provide you with guidance and code review in return.  If so, tick any of the relevant boxes below:

- [ ] I'd like to try fixing this scraper myself
- [ ] I'd like guidance to help me develop a fix
- [x] I'd prefer if the `recipe-scrapers` team try to fix this

**Comment by jayaddison** [comment]
@smilerz Thanks for the issue report and example link!

If I understand correctly, to fix this we might have to update the SpruceEats scraper so that it can attempt different ways of retrieving the ingredient list.  Does that sound right to you?

(cc @khushhalm)

**Comment by smilerz** [comment]
I think so - if Method 1 returns empty list then Method 2?

**Comment by khushhalm** [comment]
Hey, @jayaddison I'll check the SpruceEats scraper and will update you here.

**Comment by jayaddison** [comment]
@smilerz @khushhalm Thanks both :)

Yes, one approach would be to attempt one type of retrieval, and then attempt the second if the first one returns an empty list.

Another possibility would be to try to determine which type of retrieval is going to work for the current page, and then run the code for that retrieval (without any fallback if the results are empty).

It'd be nice if we can add tests that cover both situations.  To help with that, we have one existing scraper - `FarmhouseDelivery` - that provides an example of a scraper with two test suites ([one](https://github.com/hhursev/recipe-scrapers/blob/cdc9e33dd719a943e2f244b67f4ded5e3be0c9ba/tests/test_farmhousedelivery_1.py), [two](https://github.com/hhursev/recipe-scrapers/blob/cdc9e33dd719a943e2f244b67f4ded5e3be0c9ba/tests/test_farmhousedelivery_2.py)).

**Comment by khushhalm** [comment]
Hey @jayaddison, 
The issue is that the list item's class name can vary between 2 names depending on which type of recipe it is.

Proposed Solution:

1. What if I find ingredients in both the classes? `find_all("li", {"class": ["structured-ingredients__list-item", "ingredient"]})`
2. I remove the class restriction and just do `.final_all("li")`

The issue with: 
1st solution might be that the recipe has both the classes (which I think would not happen).
For 2nd solution, the issue might be there are extra items on the list other than ingredients. (Again I don't think this will happen)

If these solutions are error prone we can go with checking list item with first class and then check if the ingredients list is empty or not. If it is empty we will scrape list items with second class.

What do you suggest?

**Comment by jayaddison** [comment]
@khushhalm I prefer your first suggestion; although the issues you mention in both cases seem unlikely, I think there's a higher likelihood that an unrelated `li` element would appear on the page than that we'd find a mixture of the two ingredient CSS classes within a page -- so let's use your CSS filters from (1) to keep the results precise.

It could be helpful to add a comment to the code as well to explain why there are two classes in the `find_all` list.

**Comment by khushhalm** [comment]
Ok thanks. Will do this and create a pull request.

**Comment by jayaddison** [comment]
Resolved by #385.
